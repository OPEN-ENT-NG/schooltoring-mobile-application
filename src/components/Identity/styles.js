import { StyleSheet } from "react-native";

import { COLORS } from "../../styles/common";

const text = { color: COLORS.TEXT };

export default StyleSheet.create({
  top: {
    flexDirection: "row",
    flex: 1
  },
  topRight: {
    marginLeft: 15,
    justifyContent: "space-evenly",
    flexGrow: 1,
    flex: 1
  },
  name: {
    ...text,
    fontWeight: "bold",
    fontSize: 26
  },
  titleText: {
    ...text,
    fontSize: 20
  },
  avatarContainer: {
    padding: 10
  }
});
