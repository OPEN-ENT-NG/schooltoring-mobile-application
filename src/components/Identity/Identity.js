import React from "react";
import { View, Text } from "react-native";
import PropTypes from "prop-types";

import Avatar from "../Avatar/Avatar";

import styles from "./styles";

const Identity = props => (
  <View style={styles.top}>
    <View style={styles.avatarContainer}>
      <Avatar
        size={props.avatar.size}
        src={`${global.config.auth.endpoint}${props.avatar.src}`}
      />
    </View>

    <View style={styles.topRight}>
      <Text style={styles.name} id="username-field">
        {props.userinfo.username}
      </Text>
      <Text style={styles.titleText} id="classname-field">
        {props.userinfo.classNames[0].split("$")[1]}
      </Text>
    </View>
  </View>
);

Identity.propTypes = {
  avatar: PropTypes.shape({
    size: PropTypes.number.isRequired,
    src: PropTypes.string.isRequired
  }),
  userinfo: PropTypes.shape({
    username: PropTypes.string.isRequired,
    classNames: PropTypes.array.isRequired
  })
};

export default Identity;
