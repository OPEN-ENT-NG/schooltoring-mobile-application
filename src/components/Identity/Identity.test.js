import React from "react";
import { shallow } from "enzyme";

import Identity from "./Identity";

import state from "../../jest/state.json";

const config = { auth: { endpoint: "http://ent3-core.gdapublic.fr" } };

beforeAll(() => {
  global.config = config;
});

let rendered;

describe("Render", () => {
  test("It should render without crashing", () => {
    rendered = shallow(
      <Identity
        userinfo={state.userinfo}
        avatar={{ size: 120, src: state.userinfo.avatar }}
      />
    );

    expect(rendered).toMatchSnapshot();
  });

  test(`Username and Class name should be respectively '${
    state.userinfo.username
  }' and '${state.userinfo.classNames[0].split("$")[1]}'`, () => {
    const usernameText = rendered.find("Text").find('[id="username-field"]');
    const classnameText = rendered.find("Text").find('[id="classname-field"]');

    expect(usernameText.contains(state.userinfo.username)).toBeTruthy();
    expect(
      classnameText.contains(state.userinfo.classNames[0].split("$")[1])
    ).toBeTruthy();
  });

  test(`Avatar source should be '${config.auth.endpoint}${
    state.userinfo.avatar
  }'`, () => {
    expect(
      rendered
        .find("Avatar")
        .dive()
        .find("Image")
        .props().source.uri
    ).toEqual(`${global.config.auth.endpoint}${state.userinfo.avatar}`);
  });
});
