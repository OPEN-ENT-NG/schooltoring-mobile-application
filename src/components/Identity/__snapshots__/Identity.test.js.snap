// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render It should render without crashing 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "row",
    }
  }
>
  <View
    style={
      Object {
        "padding": 10,
      }
    }
  >
    <Avatar
      size={120}
      src="http://ent3-core.gdapublic.fr/workspace/document/49a54de3-51f8-464b-81bb-a377f508f382"
    />
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "flexGrow": 1,
        "justifyContent": "space-evenly",
        "marginLeft": 15,
      }
    }
  >
    <Text
      id="username-field"
      style={
        Object {
          "color": "#232323",
          "fontSize": 26,
          "fontWeight": "bold",
        }
      }
    >
      John DOE
    </Text>
    <Text
      id="classname-field"
      style={
        Object {
          "color": "#232323",
          "fontSize": 20,
        }
      }
    >
      2DE4
    </Text>
  </View>
</View>
`;
